File: lab4/src/container/FlatsContainer.java
package container;

import java.util.List;

import data.Flat;

public interface FlatsContainer {

    /**
     * Поиск по площади по заданным границам
     */
    List<Flat> findByArea(double lowerArea, double upperArea);

    /**
     * Добавление новой Flat
     * @param flat
     */
    void add(Flat flat);
}

File: lab6/src/container/FlatsContainer.kt
package container

import data.Flat

interface FlatsContainer {
    /**
     * Поиск по площади по заданным границам
     */
    fun findByArea(lowerArea: Double, upperArea: Double): List<Flat>

    /**
     * Добавление новой Flat
     * @param flat
     */
    fun add(flat: Flat)
}
